src/main/java/org/openrewrite/jenkins/github/AddSecurityScanWorkflow.java

```diff
@@ -0,0 +1,171 @@
+            @Override
+            public Tree visit(@Nullable Tree tree, ExecutionContext executionContext, Cursor parent) {
+                SourceFile sourceFile = (SourceFile) requireNonNull(tree);
+                if (path.toString().equals(sourceFile.getSourcePath().toString())) {
+                    found.set(true);
+                }
+                return sourceFile;
+            }
+        };
+    }
+
+    @Override
+    public Collection<? extends SourceFile> generate(AtomicBoolean found, ExecutionContext ctx) {
+        if (found.get()) {
+            return Collections.emptyList();
+        }
+        YamlParser parser = new YamlParser();
+        @Language("yml")
+        String workflow = defaultWorkflow();
+        return parser.parse(workflow)
+                .map(brandNewFile -> (Yaml.Documents) brandNewFile.withSourcePath(Paths.get(FILE_PATH)))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(AtomicBoolean found) {
+        return new YamlIsoVisitor<ExecutionContext>() {
+            @Override
+            public Yaml.Documents visitDocuments(Yaml.Documents documents, ExecutionContext executionContext) {
+                if (branches != null) {
+                    doAfterVisit(new ReplaceSequenceVisitor(BRANCHES_KEY, branches));
+                }
+                if (javaVersion != null) {
+                    doAfterVisit(new ChangePropertyValue(
+                            JAVA_VERSION_KEY,
+                            String.valueOf(javaVersion),
+                            null,
+                            null,
+                            false
+                    ).getVisitor());
+                }
+                if (buildTool != null) {
+                    doAfterVisit(new ChangePropertyValue(
+                            BUILD_TOOL_KEY,
+                            buildTool,
+                            null,
+                            null,
+                            false
+                    ).getVisitor());
+                }
+                return super.visitDocuments(documents, executionContext);
+            }
+        };
+    }
+
+    private static String defaultWorkflow() {
+        try (InputStream is = AddSecurityScanWorkflow.class.getResourceAsStream(DEFAULT_WORKFLOW_PATH)) {
+            requireNonNull(is);
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(is, UTF_8))) {
+                List<String> wanted = new LinkedList<>();
+                List<String> lines = br.lines().collect(Collectors.toList());
+                boolean licenseHeaderDone = false;
+                for (String line : lines) {
+                    if (licenseHeaderDone) {
+                        wanted.add(line);
+                    } else if (line.isEmpty()) {
+                        licenseHeaderDone = true;
+                    }
+                }
```

Comment:
Would it make sense to configure the license plugin to the workflow file with something like this?
```groovy
license {
    exclude "**/*.properties"
    excludes(["**/*.txt", "**/*.conf"])
}
```
Unless we go for downloading the file on the fly of course, in which case this wouldn't be necessary.

Reply:
Nice, thanks! Didn't know this was an option